data-prepper-plugins/dynamodb-source/src/main/java/org/opensearch/dataprepper/plugins/source/dynamodb/export/DataFileScheduler.java

```diff
@@ -88,23 +88,32 @@ public void run() {
         LOG.debug("Start running Data File Scheduler");
 
         while (!Thread.currentThread().isInterrupted()) {
-            if (numOfWorkers.get() < MAX_JOB_COUNT) {
-                final Optional<EnhancedSourcePartition> sourcePartition = coordinator.acquireAvailablePartition(DataFilePartition.PARTITION_TYPE);
-
-                if (sourcePartition.isPresent()) {
-                    activeExportS3ObjectConsumersGauge.incrementAndGet();
-                    DataFilePartition dataFilePartition = (DataFilePartition) sourcePartition.get();
-                    processDataFilePartition(dataFilePartition);
-                    activeExportS3ObjectConsumersGauge.decrementAndGet();
-                }
-            }
             try {
-                Thread.sleep(DEFAULT_LEASE_INTERVAL_MILLIS);
-            } catch (final InterruptedException e) {
-                LOG.info("InterruptedException occurred");
-                break;
+                if (numOfWorkers.get() < MAX_JOB_COUNT) {
+                    final Optional<EnhancedSourcePartition> sourcePartition = coordinator.acquireAvailablePartition(DataFilePartition.PARTITION_TYPE);
+
+                    if (sourcePartition.isPresent()) {
+                        activeExportS3ObjectConsumersGauge.incrementAndGet();
+                        DataFilePartition dataFilePartition = (DataFilePartition) sourcePartition.get();
+                        processDataFilePartition(dataFilePartition);
+                        activeExportS3ObjectConsumersGauge.decrementAndGet();
+                    }
+                }
```

Comment:
should we log when workers are not available ? Will this be too much logging ?

Reply:
I think it will be too much logging yeah. These threads continue running forever, so you'd see lots of logging even after all the data files are processed

Reply:
We can consider emitting metrics when threads are not available.